guacamole/src/main/java/org/apache/guacamole/rest/history/HistoryResource.java

```diff
@@ -86,8 +93,9 @@ public HistoryResource(UserContext userContext) {
      *     If an error occurs while retrieving the connection history.
      */
     @GET
-    @Path("connections")
+    @Path("connections{p:/?}{identifier : ([0-9]+)?}")
     public List<APIConnectionRecord> getConnectionHistory(
+            @PathParam("identifier") String identifier,
```

Comment:
Beware that not all identifiers are numeric. That said, I don't think this is the best place to add this. I think it would be better to continue using the existing per-connection endpoint for connection-specific history searches, but attempt to refactor `HistoryResource` such that it can be used both at the global level and per-connection / per-user.

Reply:
Ah, okay.  I think I misunderstood your response to my question in Slack - I thought you were saying the existing API endpoint should be removed in favor of the common `history` endpoint that is already filter-able.